shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1367,6 +1377,30 @@ - (void)startKeyBoardAnimation:(NSTimeInterval)duration {
       animations:^{
         // Set end value.
         [self keyboardAnimationView].frame = CGRectMake(0, self.targetViewInsetBottom, 0, 0);
+
+        // Set keyboard spring animation details (if animation is CASpringAnimation).
+        CAAnimation* keyboardAnimation =
+            [[self keyboardAnimationView].layer animationForKey:@"position"];
+        if ([keyboardAnimation isKindOfClass:[CASpringAnimation class]]) {
+          CASpringAnimation* keyboardSpringAnimation = (CASpringAnimation*)keyboardAnimation;
+          _keyboardSpringCurve.reset([[KeyboardSpringCurve alloc]
+              initWithStiffness:keyboardSpringAnimation.stiffness
+                        damping:keyboardSpringAnimation.damping
+                           mass:keyboardSpringAnimation.mass
+                initialVelocity:keyboardSpringAnimation.initialVelocity
+               settlingDuration:0.5]);
+
+          // Set DisplayLink tracking values
+          self.keyboardAnimationStartTime = fml::TimePoint().Now();
+          self.keyboardAnimationFrom = _viewportMetrics.physical_view_inset_bottom;
+          self.keyboardAnimationTo = self.targetViewInsetBottom;
+
+          // Double duration to match actual timing of spring animation
+          self.keyboardAnimationDuration = duration * 2;
+        } else {
+          // Reset to use fallback keyboard animation tracking.
+          _keyboardSpringCurve.reset();
+        }
```

Comment:
Maybe better to extra a new method to do this logic to make code here looks more better, maybe named `setupKeyboardAnimationCurveIfNeeded` O(∩_∩)O